src/control/XournalMain.cpp

```diff
@@ -392,11 +379,9 @@ auto XournalMain::run(int argc, char* argv[]) -> int {
             XojMsgBox::showErrorToUser(static_cast<GtkWindow*>(*win), msg);
         }
 
-        GFile* file = g_file_new_for_commandline_arg(optFilename[0]);
-        Path p = Path::fromGFile(file);
-        g_object_unref(file);
+        fs::path p(optFilename[0]);
 
-        if (!p.isEmpty()) {
+        if (fs::exists(p)) {
```

Comment:
Can you check out the following 10 lines again, there is a huge regression. It should something like:
```
        try {
             if (fs::exists(p)) {
                opened = control->openFile(p, openAtPageNumber);
            } else {
                opened = control->newFile("", p);
            }
        } catch(fs::filesystem_error const& fe) {
            string msg = FS(_F("Xournal++ failed to open {1} with error: {2}\n"
                           "Note that Xournal++ cannot open remote files at the moment.\n"
                           "You have to copy the file to a local directory.")) % p.u8string().c_str() % fe.what();
            XojMsgBox::showErrorToUser(static_cast<GtkWindow*>(*win), msg);
        }
```